proto/regen/data/v1alpha2/types.proto

```diff
@@ -0,0 +1,110 @@
+syntax = "proto3";
+
+package regen.data.v1alpha2;
+
+option go_package = "github.com/regen-network/regen-ledger/x/data";
+
+// ContentID specifies a hash based content identifier for a piece of data
+message ContentID {
+    // type specifies the IDType for the ID
+    IDType type = 1;
+
+    // hash represents the hash of the data based on the specified digest_algorithm and canonicalization algorithm
+    // for types other than ID_TYPE_RAW_UNSPECIFIED.
+    bytes hash = 2;
+
+    // digest_algorithm represents the hash digest algorithm.
+    DigestAlgorithm digest_algorithm = 3;
+
+    // media_type represents the MediaType for data with ID_TYPE_RAW_UNSPECIFIED. It should be left unset
+    // if type is not ID_TYPE_RAW_UNSPECIFIED.
+    MediaType media_type = 4;
+
+    // graph_canonicalization_algorithm represents the RDF graph canonicalization algorithm. It should
+    // be left unset if type is not ID_TYPE_GRAPH.
+    GraphCanonicalizationAlgorithm graph_canonicalization_algorithm = 5;
+}
+
+enum IDType {
+    // ID_TYPE_RAW_UNSPECIFIED specifies "raw" data which does not specify a deterministic, canonical encoding.
+    // Users of these hashes MUST maintain a copy of the hashed data which is preserved bit by bit.
+    // All other encodings in IDType specify a deterministic, canonical encoding allowing implementations to
+    // choose from a variety of alternative formats for transport and encoding while maintaining the guarantee
+    // that the canonical hash will not change. The media type for "raw" data used with ID_TYPE_RAW_UNSPECIFIED
+    // is defined by the MediaType enum.
+    ID_TYPE_RAW_UNSPECIFIED = 0;
+
+    // ID_TYPE_GRAPH specifies graph data that conforms to the RDF data model.
+    // The canonicalization algorithm used for an RDF graph is specified by GraphCanonicalizationAlgorithm.
+    ID_TYPE_GRAPH = 1;
+
+    // ID_TYPE_GEOGRAPHY specifies geographic data conforming to this canonicalization algorithm:
```

Comment:
why not defining the canonicalization algorithm for `ID_TYPE_GEOGRAPHY` types as part of the `ContentID` too? It could be relevant if we wanna use other canonical representations.

Reply:
I guess it could have its own canonicalization enum too...